storage/zynq7000-flash/flashdrv.c

```diff
@@ -347,58 +353,68 @@ static ssize_t _flashdrv_read(unsigned int id, addr_t offs, void *buff, size_t l
 }
 
 
+/* cache_readCb_t - wrapper for _flashdrv_read (*/
+static ssize_t _flashdrv_readCb(uint64_t offs, void *buff, size_t len, cache_devCtx_t *ctx)
+{
+	unsigned int regID = ctx->id;
+
+	return _flashdrv_read(regID, offs, buff, len);
+}
+
 /* Block device interface */
 
-static int _flashdrv_sync(unsigned int id)
+/* cache_writeCb_t */
+static int _flashdrv_writeCb(uint64_t offs, const void *buff, size_t len, cache_devCtx_t *ctx)
 {
 	uint8_t *src;
 	addr_t dst;
 	ssize_t res;
-	unsigned int i, pgNb;
+	unsigned int i, regID = ctx->id, pgNb;
 
 	size_t pageSz, sectSz;
 	const flash_cfi_t *cfi = &fdrv_common.info.cfi;
 
-	if (fdrv_common.regs[id].buffPos == 0) {
-		return EOK;
-	}
-
-	sectSz = CFI_SIZE_SECTION(cfi->regs[id].size);
+	sectSz = CFI_SIZE_SECTION(cfi->regs[regID].size);
 	pageSz = CFI_SIZE_PAGE(cfi->pageSize);
 	pgNb = sectSz / pageSz;
 
-	res = _flashdrv_sectorErase(id, fdrv_common.regs[id].start + (sectSz * fdrv_common.regs[id].sectID));
-	if (res < 0) {
-		mutexUnlock(fdrv_common.regs[id].lock);
-		return res;
-	}
-
+	res = _flashdrv_sectorErase(regID, offs);
```

Comment:
Is it safe to assume that we're only wrinting one cache line at the time? I guess it is, but then it should be clearly stated in the API that write callback's len param is always equal to cache line size.

Reply:
Yes, the callbacks are supposed to read/write exactly one cache line. I am therefore going to add remarks to documentation of libcache in _Callbacks_ section.